### ISSUE TITLE
[BUG] Crash report 0.4.3 "Something went wrong in the regulator thread"

### ISSUE THREAD

**Issue by richb-hanover** [Fail-Safe/sqm-autorate#106]
**Script Version** 0.4.3

**Describe the bug**

sqm-autorate.log shows:

```
Something went wrong in the regulator thread
/usr/lib/sqm-autorate/sqm-autorate.lua:833: bad argument #2 to 'format' (string expected, got nil)'
```
**To Reproduce**
1. Remove previous version with `remove.sh` verify files are not present
2. Install `setup.sh`
3. Edit `/etc/config/sqm-autorate` as instructed
4. Start the service.
5. `tail -f /tmp/sqm-autorate.csv` doesn't show any change to the settings
6. `tc -s qdisc` shows ~5mbps settings (20% of my up/down base rates)

**Expected behavior**
Should Work

**OpenWrt Details (please paste your output from `cat /etc/os-release`):**
 - OS-Release: 21.02 - See attached log files
 - Connection Type + Speed:

**Additional context**

[openwrtstats-2022-01-13-074343.gz](https://github.com/Fail-Safe/sqm-autorate/files/7862829/openwrtstats-2022-01-13-074343.gz)

**Comment by jeverley** [comment]
HI @richb-hanover  please can you take a look at #99 and confirm whether this looks like the same issue? I've a proposed amendment to the logging line in that issue which you could test.

**Comment by richb-hanover** [comment]
Looks similar. I'll try your patch later this morning

**Comment by malikshi** [comment]
> Tried the patch - the service has been running for 35 minutes and controlling the speeds (It previously stopped after a few minutes).
> 
> Output of [`retrievestats.sh`](https://github.com/Fail-Safe/sqm-autorate/discussions/108) here: [openwrtstats-2022-01-13-151041.gz](https://github.com/Fail-Safe/sqm-autorate/files/7865688/openwrtstats-2022-01-13-151041.gz)

did the patch from @jeverley fixed your problem?

**Comment by richb-hanover** [comment]
Yes - It fixed the problem

**Comment by Fail-Safe** [comment]
Fixed in `testing/lua-threads` per dc7c36c9d4992d99b4f1cbb3d603bf6383656814.
